messaging/internal/service/webhook.go

```diff
@@ -0,0 +1,258 @@
+package service
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/titpetric/factory"
+
+	"github.com/crusttech/crust/internal/http"
+	"github.com/crusttech/crust/internal/store"
+	"github.com/crusttech/crust/messaging/internal/repository"
+	"github.com/crusttech/crust/messaging/types"
+	systemService "github.com/crusttech/crust/system/service"
+	systemTypes "github.com/crusttech/crust/system/types"
+)
+
+type (
+	webhook struct {
+		db     db
+		ctx    context.Context
+		client *http.Client
+
+		users   systemService.UserService
+		webhook repository.WebhookRepository
+		message repository.MessageRepository
```

Comment:
Using Message repository directly will result in circumventing message creation logic that service covers (validation, mentions, unread inc, notifying subscribers....)

This must be changed.